### DISCUSSION TITLE
Imported Guest replays point to osu account with name Guest.

### DISCUSSION THREAD

**Discussion by Vergenter** [ppy/osu#20499] [Q&A]
### Type

Game behaviour

### Bug description

NOTE: There is no problem if I use export on normal local guest replay, remove it and readd from export.
I wanted to import replays from other osu!lazer instalation.
I used python script to identify which files in  ~/.local/share/osu/files folder are replays.(old drive)
```py
def is_file_replay(filepath: str) -> bool:
    REPLAY_HEADER = b"\x00\x80\xc3\xc9\x01\x0b"
    with open(filepath, "rb") as f:
        firstline = f.readline()
        return firstline.startswith(REPLAY_HEADER)
```
I copied them to different folder and added extension `.osr`
After import all guest replays point to player [Guest](https://osu.ppy.sh/users/11666366).
As you can see on the screenshot:
![osu_2022-09-27_02-24-26](https://user-images.githubusercontent.com/37329111/192404352-e4ba984f-3d75-4d52-978e-35b9ecf5d38e.jpg)
I attached `.osr` file from screenshot. 

### Screenshots or videos

[GUEST_REPLAY.zip](https://github.com/ppy/osu/files/9650472/GUEST_REPLAY.zip)
![osu_2022-09-27_02-24-26](https://user-images.githubusercontent.com/37329111/192404352-e4ba984f-3d75-4d52-978e-35b9ecf5d38e.jpg)

### Version

2022.923.1-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/9650547/database.log)
[network.log](https://github.com/ppy/osu/files/9650548/network.log)
[performance.log](https://github.com/ppy/osu/files/9650549/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9650550/runtime.log)

**Comment by peppy** [comment]
I don't think this is a flow we can support. Can you better describe your use case?

**Comment by Vergenter** [comment]
### Use case
Import replays(and potentially beatmaps) from other osu!lazer installations.
### In my case it's hard disk failure.
1. I had hard disk failure with my osu!lazer installation.
1. I started playing osu!lazer from clean start.
1. Later I recovered data from my hard disk.
1. Now I want to merge my osu!lazer installations.
### Reasoning for my approach
Currently I don't know any other way to move thousands of replays between osu!lazer installations.
It is especially useful if *client.realm* gets corrupted and it loses track of all replays and beatmaps(they are still in *files* folder). (It happened to me twice.)

**Reply by peppy** [reply]
Could you please provide your `client.realm`? I'd sooner recover that for you than attempt to support what you're doing.

**Reply by Vergenter** [reply]
My `client.realm` [client.realm.zip](https://github.com/ppy/osu/files/9652421/client.realm.zip)
I found out that using my solution scores aren't correctly calculated. (They need to be recalculated.)

Could you explain how could I do it in different way?
I can imagine that it could look like this:
1. Find files without collision between osu!lazer installations.
2. Move files without collision to one  osu!lazer installation.
3. Update client.relam with added files.


How should I handle collisions? (Would there be any?)

**Reply by peppy** [reply]
I can't give advice on how to do what you are trying to do because it's not something we intend to support (and figuring out what would be required takes quite a bit of thought and time), so I'm afraid you're on your own if you intend to take on that task.

Thanks for your realm file. Do you have an error showing how it is corrupted? It seems to be in good shape here, I can load it fine. It has 5,862 scores in it.

**Reply by Vergenter** [reply]
That `client.realm` from above was after merging. (Clean install and import of defective `.osr` files)
### Corrupted realms:
- [client_1646519283_corrupt.realm.zip](https://github.com/ppy/osu/files/9654057/client_1646519283_corrupt.realm.zip)
- [client_1646519612_corrupt.realm.zip](https://github.com/ppy/osu/files/9654059/client_1646519612_corrupt.realm.zip)
- [client.realm.zip](https://github.com/ppy/osu/files/9654065/client.realm.zip) (Doesn't show all replays)
### `client.realm` that doesn't show all replays:
![osu_2022-09-27_11-11-36](https://user-images.githubusercontent.com/37329111/192489851-bc289a51-672d-4536-9b02-b5d31d8e615e.jpg)
But if I look into `files`
I can see 3 replays:
- `osu/files/d/dc/dc61ae0a85c7e261cca33fb9b481ca9578db529f52315290780a2d2cda1eff48`
- `osu/files/f/f9/f9e62329c841fa1fae08dc1e21c790cf7da06e45b30f02c69a641a5d13a07f8c`
- `osu/files/8/86/86950b2d455f1ab936a6f162e3cc4907f64ecb6cf8e1caccd1c6c71946b72442`

For beatmap(Hard difficulty):
- path: `osu/files/7/7f/7f636eff7a99e3f4494b4c833fc4f6b57d65d8fd0d4878b91f02d640e1c572d2`
- beatmap id: 553728
- beatmap_set_id: 158023
- map_md5 = 'ff6dd33127811f3bdabddb74d88a1126'

To make them visible in game I needed to rename them to `*.osr` and drag&drop them into osu!lazer window.
